🧱 Updated starbound egg install script

Wazbat opened the pull request:
Updated script to handle multiple or incorrectly named .pak files

# Description

As described in #1827 the starbound server's install script does not correctly symlink mods if they do not consist of a single `contents.pak` file. I have updated the install script to loop over every single .pak file in the mod's folder, and create a symbolic link for each file, with an incrementing counter on the end in case of multiple .pak files.

I have updated the script and tested it with a new server. I have not modified the exported file, so there are other changes that the panel included.

Resolves #1827 

## Checklist for all submissions

<!-- insert X into the brackets to mark it as done. You can click preview to make the links appear clickable. -->

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

parkervcp commented:
I can look at this later today.

Wazbat commented:
Cheers!

Tested it and it seems to work ok. The only thing is I did leave in a `echo $pakfile` line where it creates the symlink. Not sure if that'd be a problem or if you'd want to change that

QuintenQVD0 commented on game_eggs/steamcmd_servers/starbound/egg-starbound.json:
```diff
@@ -1,19 +1,19 @@
-        "quay.io\/parkervcp\/pterodactyl-images:ubuntu_source"
-    ],
+    "docker_images": {
+        "quay.io\/parkervcp\/pterodactyl-images:ubuntu_source": "quay.io\/parkervcp\/pterodactyl-images:ubuntu_source"
```
Can you please try if the newer images works for this: `ghcr.io/parkervcp/games:source`

gOOvER commented on game_eggs/steamcmd_servers/starbound/egg-starbound.json:
```diff
@@ -24,7 +24,7 @@
         "installation": {
-            "script": "#!\/bin\/bash\r\n# steamcmd Base Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\n# Image to install with is 'ubuntu:18.04'\r\napt -y update\r\napt -y --no-install-recommends install curl lib32gcc1 ca-certificates\r\n\r\n## just in case someone removed the defaults.\r\nif [ \"${STEAM_USER}\" == \"\" ]; then\r\n    echo -e \"steam user is not set.\\n\"\r\n    ech ...
+            "script": "#!\/bin\/bash\r\n# steamcmd Base Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\n# Image to install with is 'ubuntu:18.04'\r\napt -y update\r\napt -y --no-install-recommends install curl lib32gcc1 ca-certificates\r\n\r\n## just in case someone removed the defaults.\r\nif [ \"${STEAM_USER}\" == \"\" ]; then\r\n    echo -e \"steam user is not set.\\n\"\r\n    ech ...
             "container": "debian:buster-slim",
```
use the Debian Installer image from GHCR

Wazbat replied:
> use the Debian Installer image from GHCR

Are you refering to the `ghcr.io/parkervcp/games:source` image that @QuintenQVD0 mentioned? I'm not sure what image I should use as that one doesn't mention debian

Wazbat replied:
That, or @parkervcp could try changing it after testing my changes maybe?

QuintenQVD0 replied:
@Wazbat the install image should the GHCR debian install image. And the game image you should try the 1 I posted and if it works you should use that one

gOOvER replied:
> That, or @parkervcp could try changing it after testing my changes maybe?

i pushed a PR to your fork

Wazbat replied:
Gotcha, just gonna leave this link here for reference

Wazbat/eggs#1

Wazbat replied:
Ahhh I see what you mean now. I was a bit confused why your code review was on this line. I saw the diff and I thought you were refering to the `script` property, not the `container`

Wazbat commented:
As requested, I have updated the docker image to `ghcr.io/parkervcp/games:source`. I also removed the extra `echo` from the install script

I tested it and it appears to run ok. Not sure if there's a reduction in memory usage with this image as I didn't note down the old memory usage. Appears to sit around 508MB on a vanilla server without mods

Wazbat commented:
Merged PR from @gOOvER which updates the installation script image, as well as removing package update and install lines from the top of the install script
```diff
#!/bin/bash
# steamcmd Base Installation Script
- #
- # Server Files: /mnt/server
- # Image to install with is 'ubuntu:18.04'
- apt -y update
- apt -y --no-install-recommends install curl lib32gcc1 ca-certificates

## just in case someone removed the defaults.
if [ "${STEAM_USER}" == "" ]; then
(...)
```

gOOvER commented:
> Merged PR [Wazbat#1](https://github.com/Wazbat/eggs/pull/1) from @gOOvER which updates the installation script docker image to `ghcr.io\/parkervcp\/installers:debian`, as well as removing package update and install lines from the top of the install script
> 
> ```diff
> #!/bin/bash
> # steamcmd Base Installation Script
> - #
> - # Server Files: /mnt/server
> - # Image to install with is 'ubuntu:18.04'
> - apt -y update
> - apt -y --no-install-recommends install curl lib32gcc1 ca-certificates
> 
> ## just in case someone removed the defaults.
> if [ "${STEAM_USER}" == "" ]; then
> (...)
> ```

you dont need this, because both are in installimage allready.

Wazbat commented:
> you dont need this, because both are in installimage allready.
> 
> lib32gcc1 also changed to lib32gcc-s1

Yeah that makes complete sense. Just commenting it here for visibility.

Comparing the bash scripts when they're just a single line with multiple escaped characters is a pain haha

gOOvER commented:
what i missed is to update the installscript to newest

https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh

Wazbat commented:
What's the status on this? Don't want to let it get stale

parkervcp commented:
I hadn't had a chance to parse the scripts until now. It looks fine. I see what was changed at least.

Easiest way to compare the scripts is to pull the string and do a replace on `\/` with `/` and `\r\n` with a new line which expands them and makes them readable.

Wazbat commented:
> Easiest way to compare the scripts is to pull the string and do a replace on `\/` with `/` and `\r\n` with a new line which expands them and makes them readable.

Yep! That seems to work, though you also have to un-escape the double quotes too

I wonder if you could do a simple github action step that would post in a comment a diff view of the scripts if that line was modified. Curious if there'd be any need for it. Could be a fun project

Wazbat commented:
> I might be able to swing something on my end.

Nice, yeah that'd be cool to see.

Hardest part would be checking the diff to see if that line in an egg file was updated, but it should be doable. Another thing is keeping the comment up to date, should the author add another commit to the PR that modifies the script again, though I believe there is an activity type that would allow you to keep checking that
